Update OCIM installation documentation (#527)
open-craft/opencraft: documentation/installation.md

Vagrant uses [VirtualBox](https://www.virtualbox.org/) to create isolated
virtual machines with the developer environment set up. To provision and
configure the developer environment as needed, Vagrant uses
[Ansible](https://www.ansible.com/).

> First, make sure that Intel virtualization is enabled in your BIOS / UEFI firmware.

Reviewer: @alanoe: nit: this cannot always be Intel virtualization as AMD CPUs, Ryzen series in particular, are very popular. So it might be better to explicitly mention `Intel VT-x/AMD-V virtualization hardware extensions' instead.

```suggestion
First, make sure that Intel VT-x/AMD-v virtualization hardware extension are enabled in your BIOS / UEFI firmware.
```

Author: Done

Then, you will need to install all these tools before you can set up your development
environment:

- [Vagrant Download](https://www.vagrantup.com/downloads.html)
- [VirtualBox Download](https://www.virtualbox.org/wiki/Downloads)

Install the plugin vagrant-vbguest:

    vagrant plugin install vagrant-vbguest

> Once you have these tools installed, download the [OCIM repository]

Reviewer: ```suggestion
Once you have these tools installed, download the [Ocim repository]
```

Author: if OCIM is an acronym, why not write it in upper case?

Reviewer: @alanoe, I don't know the reason, but `Ocim` and not `OCIM` is the official spelling. I have asked this earlier and got a confirmation. For example, see the first sentence in the `README` file which mentions `OpenCraft Instance Manager (Ocim)`.

Author: That's odd. OK, I'll change my patch to be consistent with the existing nomenclature.

(https://github.com/open-craft/opencraft) if you have not already done it.

    git clone https://github.com/open-craft/opencraft ocim
    cd ocim

> Download also the [Ansible playbooks](https://github.com/open-craft/ansible-playbooks)

Reviewer: ```suggestion
Also download the [Ansible playbooks](https://github.com/open-craft/ansible-playbooks)
```

Author: is there any difference between "download also" and "also download" ?

Reviewer: @alanoe, imho, while the latter is also grammatically correct, it sounds more colloquial and uncommon than the latter (my suggestion).

Author: Done

    vagrant ssh

Inside the virtual machine, if you are setting up a development environment,
set HUEY_ALWAYS_EAGER to false in /home/vagrant/opencraft/.env. Also, create

> a super user which will be used to login in Ocim:

Reviewer: ```suggestion
a superuser account which will be used to log in to Ocim:
```

Author: Done

    make manage createsuperuser

To check if everything is set up properly, you can run ``make test.unit`` inside
your new environment.

Vagrant will set up a VirtualBox share mapping your local development directory
to `/vagrant` inside the virtual machine. Any changes you make locally will be
reflected inside the virtual machine automatically.

Vagrant will map port 5000 inside the virtual machine to port 5000 on the host.
Once you have set everything up, you will be able to access the home webpage of

> the development server at http://localhost:5000/ using your web browser. Login

Reviewer: ```suggestion
the development server at http://localhost:5000/ using your web browser. Log in
```

Author: Done

> with the previously created superuser and you will be redirected to Ocim.

Reviewer: ```suggestion
with the previously created superuser account credentials and you will then be redirected to the Ocim instances page.
```

Author: Done

    vagrant ssh

> Inside the virtual machine, if you are setting up a development environment,

Reviewer: @alanoe, the vagrant VM and the playbooks that run on it, in particular, are only for setting up a development environment.

Author: Done

set HUEY_ALWAYS_EAGER to false in /home/vagrant/opencraft/.env. Also, create
a superuser account which will be used to log in to Ocim:

    make manage createsuperuser

To check if everything is set up properly, you can run ``make test.unit`` inside
your new environment.

Vagrant will set up a VirtualBox share mapping your local development directory
to `/vagrant` inside the virtual machine. Any changes you make locally will be
reflected inside the virtual machine automatically.

Vagrant will map port 5000 inside the virtual machine to port 5000 on the host.
Once you have set everything up, you will be able to access the home webpage of
the development server at http://localhost:5000/ using your web browser. Log in
with the previously created superuser account credentials and you will then be

> redirected to the Ocim webpage which lists the OpenEDX instances.

Reviewer: ```suggestion
redirected to the Ocim webpage which lists the Open edX instances.
```

Author: Again this is a pretty odd case, but I'll just be consistent and fix it.

    vagrant ssh

Inside the virtual machine, if you are setting up a development environment,
set HUEY_ALWAYS_EAGER to false in /home/vagrant/opencraft/.env.

Inside the virtual machine, create a super user which will be used to login

> in OCIM:

Reviewer: ```suggestion
Create a superuser account which will be used to log in to Ocim:
```

    make manage createsuperuser

To check if everything is set up properly, you can run ``make test.unit`` inside
your new environment.

Vagrant will set up a VirtualBox share mapping your local development directory
to `/vagrant` inside the virtual machine. Any changes you make locally will be
reflected inside the virtual machine automatically.

Vagrant will map port 5000 inside the virtual machine to port 5000 on the host.
Once you have set everything up, you will be able to access the home webpage of
the development server at http://localhost:5000/ using your web browser. Login at
http://localhost:5000/admin with the previously created superuser and go back to

> the home page in order to access OCIM.

Reviewer: @alanoe, why does the user have to log in via the Django admin interface? The regular login form at `/registration/login/` works just fine.

```suggestion
the home page in order to access Ocim.
```

Author: > @alanoe, why does the user have to log in via the Django admin interface? The regular login form at `/registration/login/` works just fine.

I thought that the superuser created by "make manage superuser" was only valid for the Django admin UI. I'll change the doc then.
